### PR TITLE
couch sql migration more diffs

### PR DESCRIPTION
**Pull request by snopoke**
@czue 
buddy @calellowitz 
review per commit :fishing_pole_and_fish:

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/couch_sql_migration/couchsqlmigration.py, line 26 - F401 'chunked' imported but unused
* corehq/apps/couch_sql_migration/diff.py, line 250 - E501 line too long (125 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 264 - W391 blank line at end of file
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 156 - E501 line too long (126 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 157 - E501 line too long (130 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_migration.py, line 314 - E501 line too long (135 > 115 characters)
* corehq/form_processor/models.py, line 759 - E123 closing bracket does not match indentation of opening bracket's line

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/couch_sql_migration/couchsqlmigration.py, line 26 - F401 'chunked' imported but unused
* corehq/apps/couch_sql_migration/diff.py, line 250 - E501 line too long (125 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 264 - W391 blank line at end of file
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 156 - E501 line too long (126 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 157 - E501 line too long (130 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_migration.py, line 314 - E501 line too long (135 > 115 characters)
* corehq/form_processor/models.py, line 759 - E123 closing bracket does not match indentation of opening bracket's line

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/couch_sql_migration/couchsqlmigration.py, line 26 - F401 'chunked' imported but unused
* corehq/apps/couch_sql_migration/diff.py, line 250 - E501 line too long (125 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 264 - W391 blank line at end of file
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 156 - E501 line too long (126 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 157 - E501 line too long (130 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_migration.py, line 314 - E501 line too long (135 > 115 characters)
* corehq/form_processor/models.py, line 759 - E123 closing bracket does not match indentation of opening bracket's line

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/couch_sql_migration/couchsqlmigration.py, line 26 - F401 'chunked' imported but unused
* corehq/apps/couch_sql_migration/diff.py, line 250 - E501 line too long (125 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 264 - W391 blank line at end of file
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 156 - E501 line too long (126 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 157 - E501 line too long (130 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_migration.py, line 314 - E501 line too long (135 > 115 characters)
* corehq/form_processor/models.py, line 759 - E123 closing bracket does not match indentation of opening bracket's line

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/couch_sql_migration/couchsqlmigration.py, line 26 - F401 'chunked' imported but unused
* corehq/apps/couch_sql_migration/diff.py, line 250 - E501 line too long (125 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 264 - W391 blank line at end of file
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 156 - E501 line too long (126 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 157 - E501 line too long (130 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_migration.py, line 314 - E501 line too long (135 > 115 characters)
* corehq/form_processor/models.py, line 759 - E123 closing bracket does not match indentation of opening bracket's line

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/couch_sql_migration/couchsqlmigration.py, line 26 - F401 'chunked' imported but unused
* corehq/apps/couch_sql_migration/diff.py, line 250 - E501 line too long (125 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 264 - W391 blank line at end of file
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 156 - E501 line too long (126 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 157 - E501 line too long (130 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_migration.py, line 314 - E501 line too long (135 > 115 characters)
* corehq/form_processor/models.py, line 759 - E123 closing bracket does not match indentation of opening bracket's line

**Comment by dimagimon** [comment]
There are 7 errors:

* corehq/apps/couch_sql_migration/couchsqlmigration.py, line 26 - F401 'chunked' imported but unused
* corehq/apps/couch_sql_migration/diff.py, line 251 - E501 line too long (125 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 265 - W391 blank line at end of file
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 156 - E501 line too long (126 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 157 - E501 line too long (130 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_migration.py, line 314 - E501 line too long (135 > 115 characters)
* corehq/form_processor/models.py, line 759 - E123 closing bracket does not match indentation of opening bracket's line

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/couch_sql_migration/couchsqlmigration.py, line 26 - F401 'chunked' imported but unused
* corehq/apps/couch_sql_migration/diff.py, line 240 - E501 line too long (119 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 257 - E501 line too long (125 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 271 - W391 blank line at end of file
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 156 - E501 line too long (126 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 157 - E501 line too long (130 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_migration.py, line 315 - E501 line too long (135 > 115 characters)
* corehq/form_processor/models.py, line 759 - E123 closing bracket does not match indentation of opening bracket's line

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/couch_sql_migration/couchsqlmigration.py, line 26 - F401 'chunked' imported but unused
* corehq/apps/couch_sql_migration/diff.py, line 240 - E501 line too long (119 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 261 - E501 line too long (125 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 275 - W391 blank line at end of file
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 156 - E501 line too long (126 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 157 - E501 line too long (130 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_migration.py, line 315 - E501 line too long (135 > 115 characters)
* corehq/form_processor/models.py, line 759 - E123 closing bracket does not match indentation of opening bracket's line

**Comment by czue** [comment]
fyi planning on taking a look at this over the weekend - sorry to be slow

**Comment by snopoke** [comment]
👍thanks. It's basically a long list of tweaks and edge cases

File: corehq/apps/couch_sql_migration/tests/test_migration.py [hunk 1]
```diff
@@ -112,6 +112,26 @@ def test_error_form_migration(self):
         self.assertEqual(1, len(self._get_form_ids('XFormError')))
         self._compare_diffs([])
 
+    def test_error_with_normal_doc_type_migration(self):
```
Comment (czue) [inline corehq/apps/couch_sql_migration/tests/test_migration.py]: wow this is an obscure one

File: corehq/util/test_utils.py [hunk 2]
```diff
@@ -68,25 +68,16 @@ def __exit__(self, exc_type, err, tb):
             raise SkipTest("{}{}: {}".format(msg, type(err).__name__, err))
 
 
-def softer_assert(func=None):
+class softer_assert(ContextDecorator):
```
Comment (czue) [inline corehq/util/test_utils.py]: nice

File: corehq/apps/couch_sql_migration/couchsqlmigration.py [hunk 3]
```diff
@@ -91,7 +91,9 @@ def _migrate_form_and_associated_models(self, couch_form):
             filter_form_diffs(couch_form.doc_type, diffs)
         )
 
-        case_stock_result = _get_case_and_ledger_updates(self.domain, sql_form)
+        case_stock_result = None
+        if couch_form.initial_processing_complete:
```
Comment (czue) [inline corehq/apps/couch_sql_migration/couchsqlmigration.py]: also fun

**Review by czue — APPROVED** [review]
this is looking good to me. thanks for breaking it up so cleanly.

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/couch_sql_migration/couchsqlmigration.py, line 26 - F401 'chunked' imported but unused
* corehq/apps/couch_sql_migration/diff.py, line 240 - E501 line too long (119 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 261 - E501 line too long (125 > 115 characters)
* corehq/apps/couch_sql_migration/diff.py, line 275 - W391 blank line at end of file
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 159 - E501 line too long (126 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_diffs.py, line 160 - E501 line too long (130 > 115 characters)
* corehq/apps/couch_sql_migration/tests/test_migration.py, line 315 - E501 line too long (135 > 115 characters)
* corehq/form_processor/models.py, line 759 - E123 closing bracket does not match indentation of opening bracket's line

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**